packages/bot-web-ui/src/pages/dashboard/announcements/__tests__/announcements.spec.tsx

```diff
@@ -0,0 +1,126 @@
+        xml: {
+            textToDom: (xmlString: string) => {
+                const parser = new DOMParser();
+                return parser.parseFromString(xmlString, 'text/xml');
+            },
+        },
+    },
+    Xml: { domToText: () => ({}) },
+};
+
+const mockHandleTabChange = jest.fn();
+
+describe('Announcements', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+    const mock_store = mockStore({});
+
+    beforeAll(() => {
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('should decrease the indicator count and remove it, the list of announcements should be displayed, and redirect to tutorial page upon clicking on the announcement item.', async () => {
+        const { container } = render(<Announcements handleTabChange={mockHandleTabChange} is_mobile={true} />, {
+            wrapper,
+        });
+        const button = screen.getByTestId('btn-announcements');
+        userEvent.click(button);
+
+        await waitFor(() => {
+            // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+            const notification_button = container.querySelector('.notification__button');
+
+            expect(notification_button).toBeInTheDocument();
+
+            if (notification_button) {
+                userEvent.click(notification_button);
+            }
+        });
+        await waitFor(() => {
+            expect(screen.queryByTestId('announcements__amount')).not.toBeInTheDocument();
+
+            const button_cancel = screen.getByRole('button', { name: /Learn more/i });
+            userEvent.click(button_cancel);
+        });
+        await waitFor(() => {
+            expect(mock_DBot_store?.dashboard.setActiveTab(DBOT_TABS.TUTORIAL));
+        });
+    });
+
+    it('should decrease the indicator count and remove it, the list of announcements should be displayed, and redirect to bot builder page upon clicking on the announcement item.', async () => {
+        const { container } = render(<Announcements handleTabChange={mockHandleTabChange} is_mobile={true} />, {
+            wrapper,
+        });
+        const button = screen.getByTestId('btn-announcements');
+        userEvent.click(button);
+
+        await waitFor(() => {
+            // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+            const notification_button = container.querySelector('.notification__button');
+
+            expect(notification_button).toBeInTheDocument();
+
+            if (notification_button) {
+                userEvent.click(notification_button);
```

Comment:
same here https://github.com/binary-com/deriv-app/pull/16577/files#r1726662969

Reply:
replied there in the first take of the same comment on this line